src/python-common/ceph/deployment/inventory.py

```diff
@@ -0,0 +1,79 @@
+from functools import total_ordering
+
+try:
+    from typing import List, Optional, Dict, Any
+except ImportError:
+    pass  # for type checking
+
+
+class Devices(object):
+    """
+    A container for Device instances with reporting
+    """
+
+    def __init__(self, devices):
+        # type: (List[Device]) -> None
+        self.devices = devices  # type: List[Device]
+
+    def __eq__(self, other):
+        return self.to_json() == other.to_json()
+
+    def to_json(self):
+        return [d.to_json() for d in self.devices]
+
+    @classmethod
+    def from_json(cls, input):
+        # type: (List[Dict[str, Any]]) -> Devices
+        try:
+            return cls([Device.from_json(i) for i in input])
+        except TypeError as e:
+            raise Exception(str(input))
+
+
+
+class Device(object):
+    report_fields = [
+        'rejected_reasons',
+        'available',
+        'path',
+        'sys_api',
+        'lvs',
+        'human_readable_type',
+        'device_id'
+    ]
+
+    def __init__(self,
+                 path,  # type: str
+                 sys_api=None,  # type: Optional[Dict[str, Any]]
+                 available=None,  # type: Optional[bool]
+                 rejected_reasons=None,  # type: Optional[List[str]]
+                 lvs=None,  # type: Optional[List[str]]
+                 device_id=None,  # type: Optional[str]
+                 ):
+        self.path = path
+        self.sys_api = sys_api
+        self.available = available
+        self.rejected_reasons = rejected_reasons
+        self.lvs = lvs
+        self.device_id = device_id
+
+    def to_json(self):
+        return {
+            k: getattr(self, k) for k in self.report_fields
+        }
+
+    @classmethod
+    def from_json(cls, input):
```

Comment:
Not quite sure if we should take care of the `path` parameter. `Device()` raises an exception but `Device.from_json({})` works.

Reply:
Do you have a traceback?

Reply:
No. And the path will be None. I just want to know if set path to None is correct.

Reply:
Eh, well. `path` should never be none, otherwise it's not a real device?